core/domain/voiceover_domain_test.py

```diff
@@ -394,3 +394,381 @@ def test_is_both_voiceovers_empty_should_return_successfully(self) -> None:
 
         self.assertFalse(
             entity_voiceovers_object.is_both_voiceovers_empty('content_0'))
+
+
+class ExplorationVoiceArtistsLinkUnitTests(test_utils.GenericTestBase):
+    """Test for ExplorationVoiceArtistsLink domain class."""
+
+    def setUp(self) -> None:
+        super().setUp()
+
+        self.dummy_voiceover_dict: state_domain.VoiceoverDict = {
+            'filename': 'filename1.mp3',
+            'file_size_bytes': 3000,
+            'needs_update': False,
+            'duration_secs': 6.1
+        }
+
+        self.content_id_to_voiceovers_mapping = {
+            'content_id_0': {
+                'en': ('voice_artist_1', self.dummy_voiceover_dict)
+            },
+            'content_id_1': {
+                'en': ('voice_artist_1', self.dummy_voiceover_dict),
+                'es': ('voice_artist_2', self.dummy_voiceover_dict)
+            }
+        }
+
+        self.exp_voice_artists_link = (
+            voiceover_domain.ExplorationVoiceArtistsLink(
+                exp_id='exp_id_1',
+                content_id_to_voiceovers_mapping=(
+                    self.content_id_to_voiceovers_mapping))
+            )
+        self.exp_voice_artists_link.validate()
+
+    def test_validate_exp_id(self) -> None:
+        # Test exp_id must be a string.
+        # TODO(#13059): Here we use MyPy ignore because after we fully type
+        # the codebase we plan to get rid of the tests that intentionally test
+        # wrong inputs that we can normally catch by typing.
+        with self.assertRaisesRegex(
+            utils.ValidationError,
+            'Expected exploration id to be string'
+        ):
+            self.exp_voice_artists_link.exp_id = 123  # type: ignore[assignment]
+            self.exp_voice_artists_link.validate()
+
+        # Reset for next test.
+        self.exp_voice_artists_link.exp_id = 'exp_id_1'
```

Comment:
Break this down, don't test multiple test functionalities in a single test function. Keep it as granular as possible.